src/library_manager/lib_manager.c

```diff
@@ -110,79 +137,135 @@ static int lib_manager_load_module(uint32_t module_id, struct sof_man_module *mo
 	if (ext_lib->mods_exec_load_cnt == 1) {
 		struct sof_man_module *module_entry =
 		    (struct sof_man_module *)((char *)desc + SOF_MAN_MODULE_OFFSET(0));
+
 		for (size_t idx = 0; idx < desc->header.num_module_entries;
 				++idx, ++module_entry) {
 			if (module_entry->type.lib_code) {
-				ret = lib_manager_load_module(lib_id << LIB_MANAGER_LIB_ID_SHIFT |
-							      idx, mod, desc);
+				int ret = lib_manager_load_module(lib_id << LIB_MANAGER_LIB_ID_SHIFT |
+								  idx, mod, desc);
 				if (ret < 0)
-					goto err;
+					return ret;
```

Comment:
@softwarecki It should be freed, yes, but I just moved the code from below. And I think some other PRs will rewrite this anyway. So I just moved it out of the way for now to let those other PRs work on this.

Reply:
...and yes, I meant #8544 by @softwarecki

Reply:
In that case, let's not modify the code that will be changed by another PR. There is no point in introducing a merge conflicts in another PR.

Reply:
In that case, let's not modify the code that will be changed by another PR. There is no point in introducing a merge conflicts in another PR.

Reply:
In that case, let's not modify the code that will be changed by another PR. There is no point in introducing a merge conflicts in another PR.

Reply:
@softwarecki I cannot and shall not sit and wait because I know that somebody else's work is in progress. If your PR gets merged first - I'll update mine, if mine is merged first - you'll update yours